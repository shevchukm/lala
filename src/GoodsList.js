import React, { Component } from 'react';
import Axios from 'axios';
import AddGood from './AddGood';

class GoodsList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            goods: [],
            search: '',
            isStocked: false,
            sort: 'name'
        };
    }
  
    componentDidMount() {
        this.getGoods();
    }

    getGoods = () =>{
        Axios.get('http://localhost:3012/goods')
            .then(res =>{
                this.setState({goods: res.data});
            });
    }
  
    handleFilters = (e) => {
        e.target.type === 'checkbox' ?
            this.setState({[e.target.name]: e.target.checked}) : this.setState({[e.target.name]: e.target.value});
    }

    sortOrder = (a, b) => {
        return this.state.sort === 'price' ?
            +a.price.slice(1) > +b.price.slice(1) : a[this.state.sort] > b[this.state.sort];
    
    }

    handleClick = () => {
        this.setState({search: ''});
    }

    render() {
        let filteredGoods = this.state.goods.filter((good) =>
            this.state.isStocked ?
                good.name.toLowerCase().includes(this.state.search.toLowerCase()) && good.stocked === false :
                good.name.toLowerCase().includes(this.state.search.toLowerCase())
        );

        filteredGoods = filteredGoods.sort(this.sortOrder).map((good, index) =>
            good.stocked ?
                <li key={index}> {good.name} -- {good.price} </li>
                :<li key={index} 
                    style={{color: 'red'}}>
                    {good.name} -- {good.price} </li>
        );

        return(
            <div>
                search
                <input
                    value={this.state.search}
                    name = "search"
                    onChange={this.handleFilters}
                    type="text" />
                <button
                    onClick = {this.handleClick} > reset </button>
                stoked:
                <input
                    type="checkbox"
                    name = "isStocked"
                    checked={this.state.isStocked}
                    onChange={this.handleFilters} />
                <p>sort by </p>
                <select 
                    name = "sort"
                    onChange={this.handleFilters}
                    value={this.state.sort}>
                    <option value="name">name</option>
                    <option value="price">price</option>
                </select>

                <ul>
                    {filteredGoods}
                </ul>
                
                <AddGood update={this.getGoods} />
            </div>
        );
    }
}

export default GoodsList;
