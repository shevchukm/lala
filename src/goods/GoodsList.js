import React, { Component } from 'react';
import {getGoods, deleteGoods } from '../actions/actionsGoods';
import AddGood from './AddGood';
import ActionBar from './ActionBar';

class GoodsList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            goods: [],
            search: '',
            isStocked: false,
            sort: 'name'
        };

    }
  
    componentDidMount() {
        getGoods().then(res => this.setState({goods: res.data}));
    }
  
    handleFilters = (e) => {
        e.target.type === 'checkbox' ?
            this.setState({[e.target.name]: e.target.checked}) : this.setState({[e.target.name]: e.target.value});
    }

    sortOrder = (a, b) => {
        return this.state.sort === 'price' ?
            +a.price.slice(1) > +b.price.slice(1) : a[this.state.sort] > b[this.state.sort];
    
    }

    handleReset = () => {
        this.setState({search: ''});
    }

    handleDelete = (name) => {
        deleteGoods(name).then(() => {
            getGoods().then(res => this.setState({goods: res.data}));
        });
    }

    render() {
        let filteredGoods = this.state.goods.filter((good) =>
            this.state.isStocked ?
                good.name.toLowerCase().includes(this.state.search.toLowerCase()) && good.stocked === false :
                good.name.toLowerCase().includes(this.state.search.toLowerCase())
        );

        filteredGoods = filteredGoods.sort(this.sortOrder).map((good, index) =>
            good.stocked ?
                <li key={index}> {good.name} -- {good.price} 
                    <button type="button"
                     onClick={() => this.handleDelete(good.name)}> delete</button> </li>
                :<li key={index} 
                    style={{color: 'red'}}>
                    {good.name} -- {good.price}
                    <button type="button"
                     onClick={() => this.handleDelete(good.name)}> delete</button>  
                </li>
        );

        return(
            <div>
                <ActionBar
                    state={this.state}
                    handleFilters={this.handleFilters}
                    sortOrder={this.sortOrder}
                    handleReset={this.handleReset}
                />
                
                <ul>{filteredGoods}</ul>
                <AddGood update={this.getGoods} />
            </div>
        );
    }
}

export default GoodsList;
