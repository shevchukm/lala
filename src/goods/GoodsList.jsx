import React, { Component } from 'react';
import { getGoods, deleteGoods } from '../actions/actionsGoods';
import FilterBar from './FiltersBar';
import { Table, Message, Button, Container, Icon } from 'semantic-ui-react';

class GoodsList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            goods: [],
            search: '',
            isStocked: false,
            sort: 'name',
            error: ''
        };

    };
  
    componentDidMount() {
        this.getGoods();
    };

    getGoods = () => {
        getGoods().then(res => this.setState({goods: res.data, error: ''}))
        .catch(err => this.setState({ error: err }));
    };
  
    handleFilters = (event, data) => {
        const { name, value, type } = data;
        if (type === 'checkbox') {
            this.setState({ [name]: !this.state[name] });
        } else { this.setState({ [name]: value });}
    };

    sortOrder = (a, b) => {
        return this.state.sort === 'price' 
            ? +a.price.slice(1) > +b.price.slice(1) 
            : a[this.state.sort] > b[this.state.sort];
    
    };

    handleReset = () => {
        this.setState({search: ''});
    };

    handleDelete = (name) => {
        deleteGoods(name).then(() => { this.getGoods(); })
        .catch(err => this.setState({ error: err.response.data }));
    };

    render() {
        let filteredGoods = this.state.goods.filter((good) =>
            this.state.isStocked ?
                good.name.toLowerCase().includes(this.state.search.toLowerCase()) && good.stocked === false :
                good.name.toLowerCase().includes(this.state.search.toLowerCase())
        );

        filteredGoods = filteredGoods.sort(this.sortOrder).map((good, index) =>
                <Table.Row key={index}>
                    <Table.Cell>
                        {good.name}  
                    </Table.Cell >
                    <Table.Cell>
                        {good.price}
                    </Table.Cell>
                    <Table.Cell>
                        {good.category}
                    </Table.Cell>
                    <Table.Cell>
                        <Button type="button"
                                onClick={() => this.handleDelete(good.name)}
                                icon
                                color="red">
                            <Icon name='delete' />
                        </Button>
                    </Table.Cell>
                </Table.Row >

        );

        return(
            <Container>
                <FilterBar
                    state={this.state}
                    onHandleFilters={this.handleFilters}
                    sortOrder={this.sortOrder}
                    handleReset={this.handleReset}
                />
                
                <Table basic='very' celled collapsing>
                    <Table.Header>
                    <Table.Row>
                        <Table.HeaderCell>Name</Table.HeaderCell>
                        <Table.HeaderCell>Price</Table.HeaderCell>
                        <Table.HeaderCell>Delete</Table.HeaderCell>
                        <Table.HeaderCell>Category</Table.HeaderCell>
                    </Table.Row>
                    </Table.Header>

                    <Table.Body>
                    {filteredGoods}
                    </Table.Body>
                </Table>
                { this.state.error &&
                <Message warning compact>
                    <Message.Header>Error has occured</Message.Header>
                    <p>{this.state.error.toString()}</p>
                    <Button onClick={this.getGoods}>try again</Button>
                </Message>
                }
            </Container>
        );
    }
}

export default GoodsList;
