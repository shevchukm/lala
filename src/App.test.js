import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
Enzyme.configure({adapter: new Adapter()});

import GoodsList from './GoodsList';



test('button clicks', () => {
  // const component = renderer.create(
  //   <GoodsList />,
  // );
  // let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  const mylist = shallow(<GoodsList />)
  mylist.find('button').simulate('click');
  expect(mylist.state().search).toEqual('misha');
});
