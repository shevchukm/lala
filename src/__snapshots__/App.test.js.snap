// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button clicks 1`] = `
<div>
  search
  <input
    name="search"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
     reset 
  </button>
  stoked:
  <input
    checked={false}
    name="isStocked"
    onChange={[Function]}
    type="checkbox"
  />
  <p>
    sort by 
  </p>
  <select
    name="sort"
    onChange={[Function]}
    value="name"
  >
    <option
      value="name"
    >
      name
    </option>
    <option
      value="price"
    >
      price
    </option>
  </select>
  <ul />
  <div>
    category:
    <input
      name="category"
      onChange={[Function]}
      type="text"
      value=""
    />
    price:
    <input
      name="price"
      onChange={[Function]}
      type="text"
      value=""
    />
    stocked:
    <input
      checked={false}
      name="stocked"
      onChange={[Function]}
      type="checkbox"
    />
    name:
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      onClick={[Function]}
      type="button"
      value="add good"
    />
  </div>
</div>
`;
